src/clr-angular/forms/combobox/option.ts

```diff
@@ -3,71 +3,94 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, ElementRef, HostBinding, HostListener, Inject, Input, OnDestroy, Optional } from '@angular/core';
-import { Subscription } from 'rxjs';
+import { Component, ElementRef, HostBinding, HostListener, Inject, Input, Optional, OnInit } from '@angular/core';
 
 import { POPOVER_HOST_ANCHOR } from '../../popover/common/popover-host-anchor.token';
-import { ClrPopoverToggleService } from '../../utils/popover/providers/popover-toggle.service';
 
 import { OptionSelectionService } from './providers/option-selection.service';
-
+import { ComboboxFocusHandler, OptionData as OptionProxy } from './providers/combobox-focus-handler.service';
+import { UNIQUE_ID, UNIQUE_ID_PROVIDER } from '../../utils/id-generator/id-generator.service';
 @Component({
   selector: 'clr-option',
-  templateUrl: './option.html',
-  host: { '[class.clr-option]': 'true' },
+  template: `
+    <ng-content></ng-content>
+  `,
+  providers: [UNIQUE_ID_PROVIDER],
+  host: {
+    '[class.clr-option]': 'true',
+    '[attr.role]': '"option"',
+    // Do not remove. Or click-selection will not work.
+    '[attr.tabindex]': '-1',
+    '[attr.id]': 'optionId',
+  },
 })
-export class ClrOption<T> implements OnDestroy {
-  private subscription: Subscription;
+export class ClrOption<T> implements OnInit {
+  // A proxy with only the necessary data to be used for a11y and the focus handler service.
+  public optionProxy: OptionProxy<T> = new OptionProxy(null, null);
+
+  private _id: string;
+  @Input('id')
+  set optionId(id: string) {
+    this._id = id;
+    this.optionProxy.id = this._id;
+  }
+  get optionId() {
+    return this._id;
+  }
 
-  @HostBinding('class.active') selected = false;
+  private _value: T;
+  @Input('clrValue')
+  set value(value: T) {
+    this._value = value;
+    this.optionProxy.value = value;
+  }
+  get value(): T {
+    return this._value;
+  }
 
-  @Input('clrValue') value: T;
+  @HostBinding('class.active')
+  get selected() {
+    return (
+      this.optionSelectionService.selectionModel && this.optionSelectionService.selectionModel.containsItem(this.value)
+    );
+  }
 
   constructor(
-    private toggleService: ClrPopoverToggleService,
     @Optional()
```

Comment:
Should this really be optional? We throw an error below if its not available.

Reply:
Legacy code, removed! Thanks